doc/cookbook/namedRoutes/NamedRoutes.lhs

```diff
@@ -0,0 +1,373 @@
+
+```
+  "version" -> Get Version
+  /
+api          "list" -> Get [Movie] ?sortBy= Title | Year       (sort by the Title or the Year)
+  \        /
+  "movies"                Get Movie
+           \                /
+            Capture MovieId - Put Movie
+                            \
+                              Delete MovieId
+```
+
+In this example, we create a very simple endpoint for the Version,
+and several complex endpoints that use nested records for the CRUD part of the movie.
+
+So, the URLs would look like
+
+- GET …/version
+- GET …/movies/list?sortby=Title
+- GET …/movies/<MovieId>/
+- PUT …/movies/<MovieId>/
+- DELETE …/movies/<MovieId>
+
+### API Type
+
+Now that we have a very clear idea of the API we want to make, we need to transform it into usable Haskell code:
+
+``` haskell
+
+data API mode = API
+    { version :: mode :- "version" :> Get '[JSON] Version
+    , movies :: mode :- "movies" :> NamedRoutes MoviesAPI
+    } deriving stock Generic
+
+type Version = String -- This will do for the sake of example.
+
+```
+Here, we see the first node of our tree. It contains the two branches “version” and “movies” respectively:
+
+The “version” branch is very simple and self-explanatory.
+The “movies” branch will contain another node, represented by another record (see above). That is why we need the `NameRoutes` helper.
+
+Note:
+
+The `mode` type parameter indicates into which implementation the record’s `Generic` representation will be transformed—as a client or as a server. We will discuss that later.
+
+Let's jump into the "movies" subtree node:
+
+
+``` haskell
+
+data MoviesAPI mode = MoviesAPI
+    { list :: mode :- "list" :> QueryParam "SortBy" SortBy :> Get '[JSON] [Movie]
+    , movie :: mode :- Capture "movieId" MovieId :> NamedRoutes MovieAPI
+    } deriving stock Generic
+
+data SortBy = Year | Title
+
+instance ToHttpApiData SortBy where
+  toQueryParam Year = "year"
+  toQueryParam Title = "title"
+
+instance FromHttpApiData SortBy where
+  parseQueryParam "year" = Right Year
+  parseQueryParam "title" = Right Title
+  parseQueryParam param = Left $ param <> " is not a valid value"
+
+```
+So, remember, this type represents the `MoviesAPI` node that we’ve connected earlier to the main `API` tree.
+
+In this subtree, we illustrated both an endpoint with a **query param** and also, a **capture** with a subtree underneath it.
+
+So, let's go deeper into our API tree.
+
+``` haskell
+data MovieAPI mode = MovieAPI
+  { get :: mode :- Get '[JSON] (Maybe Movie)
+  , update :: mode :- ReqBody '[JSON] Movie :> Put '[JSON] NoContent
+  , delete :: mode :- Delete '[JSON] NoContent
+  } deriving stock Generic
+```
+
+As you can see, we end up implementing the deepest routes of our API.
+
+Small detail: as our main API tree is also a record, we need the `NamedRoutes` helper.
+To improve readability, we suggest you create a type alias:
+
+``` haskell
+type MovieCatalogAPI = NamedRoutes API
```

Comment:
Why articulate everything around a top-level `NamedRoute` instead of using `genericServe` and friends? There is a way for users to just not care about "the old way" at all. Perhaps a final section could explain how to convert between "old style" and "record style" (back and forth) at the level of API types, server handlers, clients, etc (all the `fromServant`/`toServant` business). I can write that if needed.